lib/util/mail/database_delivery.rb

```diff
@@ -1,54 +1,50 @@
 # munged out of https://gist.github.com/d11wtq/1176236
 module Mail
   class DatabaseDelivery
-
     def initialize(parameters)
       @parameters = {}.merge(parameters)
     end
 
     def deliver!(mail)
-      is_html, body = content_and_type mail
-      subject       = ApplicationMailer.remove_prefix mail.subject
-      from          = mail[:from].addresses.first || ENV['DEFAULT_FROM']
+      is_html, body = content_and_type(mail)
+      subject = ApplicationMailer.remove_prefix(mail.subject)
+      from = mail[:from]&.to_s || ENV['DEFAULT_FROM']
+      Rails.logger.fatal 'no DEFAULT_FROM specified in .env; mail cannot be sent' if from.nil?
 
-      if from.nil?
-        Rails.logger.fatal "no DEFAULT_FROM specified in .env; mail cannot be sent"
-      end
-      Contact.where( email: mail[:to].addresses ).each do |contact|
+      Contact.where(email: mail[:to].addresses).each do |contact|
         # store the "email" in the database
         message = ::Message.create(
           contact_id: contact.id,
           subject: subject,
-          body:    body,
-          from:    from,
-          html:    is_html,
+          body: body,
+          from: from,
+          html: is_html,
         )
         user = contact.user
-        if user.blank? || user.continuous_email_delivery?
-          (1..10).each do
-            begin
-              ::ImmediateMailer.with(message: message, to: contact.email).immediate.deliver_now
-              message.update(sent_at: Time.current, seen_at: Time.current)
-              break
-            rescue Net::SMTPAuthenticationError
-              sleep 5
-            end
-          end
+        next unless user.blank? || user.continuous_email_delivery?
+
+        10.times do
+          ::ImmediateMailer.immediate(message, contact.email).deliver_now
+          message.update(sent_at: Time.current, seen_at: Time.current)
+          break
+        rescue Net::SMTPAuthenticationError
```

Comment:
Maybe not the time for this, but we get intermittent auth failures?

Reply:
I believe this was in response to Google Mail telling us we'd sent too many messages, but it might have been something else.